File: Java Based Archives/DI&Dagger/app/src/main/java/com/android/di/NCBatteryModule.java
package com.android.di;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module
abstract class NCBatteryModule {

    @Binds
    abstract Battery bindNCBattery(NickelCadmiumBattery nickelCadmiumBattery);

}

File: Views/DependencyInjection/app/src/main/java/com/example/dependencyInjection/NCBatteryModule.kt
package com.example.dependencyInjection

import dagger.Binds
import dagger.Module
import dagger.Provides

@Module
abstract class NCBatteryModule {
    @Binds
    abstract fun bindsNCBattery(nickelCadmiumBattery: NickelCadmiumBattery): Battery

}